src/components/Tokens/TokenDetails/PriceChart.tsx

```diff
@@ -117,11 +117,24 @@ const TimeButton = styled.button<{ active: boolean }>`
   }
 `
 
-function getTicks(startTimestamp: number, endTimestamp: number, numTicks = 5) {
-  return Array.from(
+function getTicks(startTimestamp: number, endTimestamp: number, forceStartOfMonthTicks = false, numTicks = 5) {
+  const ticks = Array.from(
     { length: numTicks },
     (v, i) => endTimestamp - ((endTimestamp - startTimestamp) / (numTicks + 1)) * (i + 1)
   )
+  if (!forceStartOfMonthTicks) return ticks
+
+  // when a tick maps to a date not on the first of the month, modify the tick to the closest
+  // first of month date. For example, Dec 31 becomes Jan 1, and Dec 5 becomes Dec 1.
+  return ticks.map((tick) => {
```

Comment:
Would it be possible to instead put this logic in monthFormatter, or create a monthTickFormatter and pass that in at line 152?

Reply:
good idea, separated out into a different function getStartOfMonthTicks to make lines 152 and 156 more readable.